Keep ProjectSettings overrides right below the original setting

dugramen opened the pull request:
<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->
resolves https://github.com/godotengine/godot-proposals/issues/11778
![image](https://github.com/user-attachments/assets/3614b316-84ee-4ec2-a179-4627653c32e3)

Commit:
Keep ProjectSettings overrides right below the original setting

Calinou reviewed (commented):
Tested locally, it works as expected. The order is preserved correctly after restarting the editor.

Before | After
-|-
![image](https://github.com/user-attachments/assets/9ef49188-3ab9-4b05-880a-9fe7142a05e7) | ![image](https://github.com/user-attachments/assets/abdf0cbf-c84c-4b49-bd90-175bdee51729)

Calinou commented on core/config/project_settings.cpp:
```diff
@@ -395,6 +395,7 @@ void ProjectSettings::_get_property_list(List<PropertyInfo> *p_list) const {
 	_THREAD_SAFE_METHOD_
 
 	RBSet<_VCSort> vclist;
+	HashMap<String, Vector<_VCSort>> hm;
```
I suggest using a more explicit name for the hashmap rather than just `hm`. The surrounding code already has a lot of 1/2-letter variable names so it's not easy to understand.

dugramen replied:
Got it. Is there a convention for it. I'm thinking something like setting_to_overrides or maybe overrides_per_setting

Calinou replied:
We don't really have a convention in place (other than `snake_case`), but using a few words like this is fine.

dugramen commented:
Updated with better variable names. Also fixed a failing test (I hope), where it didn't account for settings that already have a `.` in them. 

(P.S. Is there an easy way to run tests locally so I can see if its fixed before pushing?)

Calinou commented:
> (P.S. Is there an easy way to run tests locally so I can see if its fixed before pushing?)

Yes: https://docs.godotengine.org/en/stable/contributing/development/code_style_guidelines.html#pre-commit-hook

KoBeWi commented on core/config/project_settings.cpp:
```diff
@@ -435,23 +436,45 @@ void ProjectSettings::_get_property_list(List<PropertyInfo> *p_list) const {
+		if (dot != -1 && !custom_prop_info.has(vc.name)) {
+			StringName n = vc.name.substr(0, dot);
+			if (props.has(n)) { // Property is an override.
+				if (!setting_overrides.has(n)) {
+					Vector<_VCSort> vvc;
+					setting_overrides.insert(n, vvc);
+				}
+				setting_overrides.get(n).append(vc);
```
You can do that:
```suggestion
				setting_overrides[n].append(vc);
```
HashMap supports it.

KoBeWi reviewed (approved):
Looks ok.

The added HashMap and StringName operations might make the function slower, but it shouldn't have visible impact.

dugramen commented:
Sorry it took a bit. Just pushed the change

Repiteo commented:
Thanks!
